src/types/params/plain_secret.rs

```diff
@@ -146,7 +146,24 @@ impl<'a> PlainSecretParamsRef<'a> {
                         let mut secret = [0u8; 32];
                         secret[32 - d.len()..].copy_from_slice(d.as_bytes());
 
-                        Ok(SecretKeyRepr::ECDH(ECDHSecretKey {
+                        Ok(SecretKeyRepr::ECDH(ECDHSecretKey::Cv25519 {
+                            oid: curve.oid(),
+                            hash: *hash,
+                            alg_sym: *alg_sym,
+                            secret,
+                        }))
+                    }
+                    ECCCurve::P521 => {
+                        use cipher::Unsigned;
+                        use p521::{elliptic_curve::Curve, NistP521};
+
+                        const SECRET_SIZE: usize = <NistP521 as Curve>::FieldBytesSize::USIZE;
+                        ensure!(d.len() <= SECRET_SIZE, "invalid secret");
+
+                        let mut secret = [0u8; SECRET_SIZE];
+                        secret[SECRET_SIZE - d.len()..].copy_from_slice(d.as_bytes());
```

Comment:
I think [`slice::rchunks_exact`](https://doc.rust-lang.org/std/primitive.slice.html#method.rchunks_exact) is good for these kind of cases.

```suggestion
                        secret.rchunks_exact(d.len()).next().unwrap().copy_from_slice(d.as_bytes());
```